### PR TITLE
Fix parsing method member forms

### PR DESCRIPTION
**Pull request by davejcameron**
## Problem

Jessie already parses object method shorthand and static class members, but two valid JavaScript method forms fall through to misleading parser errors.

```js
({ "x/y.js"(exports, module) { module.exports = {} } })
class A { static m(a) { return a } }
```

Both should parse, but currently report `Unclosed {` around the method body.

## Fix

Allow the method shorthand parser to accept static string-literal property keys, preserving the existing `[':', key, ['=>', ...]]` AST shape.

Parse `static m() {}` class members as `['static', [':', 'm', ['=>', ...]]]`, matching the existing static member wrapper and method shorthand representation.

## Validation

- Direct parser tests for string-literal object methods and static class methods
- `node test/feature/async-class.js`: 23 tests, 64 assertions

I also ran `pnpm test`; it gets through the bundle/perf tests but still has the existing unicode scaling stack overflows unrelated to this change.

### CODE REVIEW

File: feature/class.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
-import { binary, unary, token, expr, space, next, parse, keyword, word, skip } from '../parse.js';
+import { binary, token, expr, space, next, parse, keyword, word, skip, cur, idx } from '../parse.js';
 import { block } from './if.js';
 
 const TOKEN = 200, PREFIX = 140, COMP = 90;
+const OPAREN = 40, CPAREN = 41, OBRACE = 123, CBRACE = 125;
 
 // static member → ['static', member]
-unary('static', PREFIX);
+token('static', PREFIX, a => {
```
Comment (dy) [inline feature/class.js:10]: Can it be kept as unary call - we tend to keep token for low-level primitives outside of normal syntax rules.
@Copilot

**Review by dy — CHANGES_REQUESTED** [review]

File: feature/class.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+  if (a) return;
+  space();
+  const name = next(parse.id);
+  if (!name) return;
+  space();
+  if (cur.charCodeAt(idx) !== OPAREN) return ['static', name];
+  skip();
+  const params = expr(0, CPAREN) || null;
```
[the remark this answers is not in the document]
Comment (dy) [reply feature/class.js:18]: @copilot apply changes based on [this feedback](https://github.com/dy/subscript/pull/34#discussion_r3235236974)
I would try using `unary` still for static - `token` is for low-level primitives

**Comment by dy** [comment]
Thanks @davejcameron — landed an alternative implementation in 3dec13d that keeps `unary('static', ...)` and routes `static m() {}` and string-literal-key methods through the existing method-shorthand handler in `accessor.js`.

Two reasons for the alternative:

1. `token('static', ...)` re-implemented method-shorthand parsing inline and broke `static #x` (private fields), since `next(parse.id)` won't consume `#`. The merged change preserves `unary('static')` (raised above ACCESS precedence so `static m` collapses to `['static', 'm']` and the outer `(` reaches the method-shorthand handler).
2. The string-literal key is kept as a literal node `[, "x/y.js"]` (matching how `{ "y": 2 }` is represented in [test/parse.js:266](https://github.com/dy/subscript/blob/main/test/parse.js#L266)) rather than unwrapped to a plain string — avoids the codegen issue Copilot flagged.
